tests/test_cuda_bd_exchange_mover.py

```diff
@@ -328,70 +375,24 @@ def test_moves_in_a_water_box(steps_per_move, moves, box_size, precision, rtol,
+    bdem = klass(N, group_idxs, params, DEFAULT_TEMP, nb.potential.beta, cutoff, seed, num_proposals_per_move, 1)
 
     ref_bdem = RefBDExchangeMove(nb.potential.beta, cutoff, params, group_idxs, DEFAULT_TEMP)
 
-    assert bdem.last_log_probability() == 0.0, "First log probability expected to be zero"
-    accepted = 0
-    last_conf = conf
-    for _ in range(moves // steps_per_move):
-        x_move, x_box = bdem.move(last_conf, box)
-        # The box will never change
-        np.testing.assert_array_equal(box, x_box)
-        num_moved = 0
-        new_pos = None
-        idx = -1
-        for i, mol_idxs in enumerate(group_idxs):
-            if not np.all(x_move[mol_idxs] == last_conf[mol_idxs]):
-                num_moved += 1
-                new_pos = x_move[mol_idxs]
-                idx = i
-        if num_moved > 0:
-            accepted += 1
-            # The molecules should all be imaged in the home box
-            np.testing.assert_allclose(image_frame(group_idxs, x_move, x_box), x_move)
-            # Verify that the probabilities and per mol energies agree when we do accept moves
-            # can only be done when we only attempt a single move per step
-            if steps_per_move == 1:
-                before_log_weights = ref_bdem.batch_log_weights(last_conf, box)
-                after_log_weights, tested = ref_bdem.batch_log_weights_incremental(
-                    last_conf, x_box, idx, new_pos, before_log_weights
-                )
-                np.testing.assert_array_equal(tested, x_move)
-                ref_log_prob = np.minimum(logsumexp(before_log_weights) - logsumexp(after_log_weights), 0.0)
-                ref_prob = np.exp(ref_log_prob)
-                assert np.isfinite(ref_prob) and ref_prob > 0.0
-                np.testing.assert_allclose(
-                    np.exp(bdem.last_log_probability()),
-                    ref_prob,
-                    rtol=rtol,
-                    atol=atol,
-                )
-        elif num_moved == 0:
-            np.testing.assert_array_equal(last_conf, x_move)
-        assert steps_per_move != 1 or num_moved <= 1, "More than one mol moved, something is wrong"
-        last_conf = x_move
-    assert bdem.n_proposed() == moves
-    if moves < 10_000:
-        assert accepted > 0, "No moves were made, nothing was tested"
-    else:
-        assert bdem.n_accepted() > 10
-        assert bdem.acceptance_fraction() >= 0.0001
-    if steps_per_move == 1:
-        np.testing.assert_allclose(bdem.acceptance_fraction(), accepted / moves)
-        assert bdem.n_accepted() == accepted
-    else:
-        assert bdem.n_accepted() >= accepted
-        assert bdem.acceptance_fraction() >= accepted / moves
+    verify_bias_deletion_moves(
+        group_idxs, bdem, ref_bdem, conf, box, total_num_proposals, num_proposals_per_move, rtol, atol
+    )
+    if total_num_proposals >= 10_000:
```

Comment:
Realized that this value of 1 move per 1000 proposals is way off. Using the bulk mc example, we load a water box with an empty space in the middle. And if you don't equilibrate it enough (>200k steps) you are left with a void which allows more moves to be accepted then you would expect in bulk. Looks to be closer to 1 accepted move per 7.5k proposals